wolfssl/ssl.h

```diff
@@ -1681,7 +1681,9 @@ WOLFSSL_API void wolfSSL_sk_CIPHER_free(WOLF_STACK_OF(WOLFSSL_CIPHER)* sk);
 WOLFSSL_API WOLFSSL_SESSION* wolfSSL_get1_session(WOLFSSL* ssl);
 
 WOLFSSL_API WOLFSSL_X509* wolfSSL_X509_new(void);
+WOLFSSL_API WOLFSSL_X509* wolfSSL_X509_new_ex(void* heap);
 WOLFSSL_API WOLFSSL_X509* wolfSSL_X509_dup(WOLFSSL_X509* x);
+WOLFSSL_API WOLFSSL_X509* wolfSSL_X509_dup_ex(WOLFSSL_X509* x, void* heap);
```

Comment:
Do you really need `wolfSSL_X509_dup_ex`? Can't the `wolfSSL_X509_dup` just use `x->heap` from the previous one? If you have places where heap isn't populated then switch those to using `wolfSSL_X509_new_ex`.

Reply:
you're right, I had this in my other PR but was following what the customer gave me for this one